index/mem.go

```diff
@@ -58,56 +48,26 @@ func (t *MemTable) open() {
 	n := len(t.table)
 	if n > 0 {
 		//XXX escaping to GC
-		t.keys = make([]zcode.Bytes, n)
-		k := 0
-		for key := range t.table {
-			t.keys[k] = []byte(key)
-			k++
+		t.values = make([]*zed.Record, 0, n)
+		for _, value := range t.table {
+			t.values = append(t.values, value)
 		}
-		compare := expr.LookupCompare(t.keyType)
-		sort.SliceStable(t.keys, func(a, b int) bool {
-			return compare(t.keys[a], t.keys[b]) < 0
+		resolvers := make([]expr.Evaluator, len(t.keys))
+		for k, key := range t.keys {
+			resolvers[k] = expr.NewDotExpr(key)
+		}
```

Comment:
Use `append` here for consistency with the handling of `t.values`.
```suggestion
		resolvers := make([]expr.Evaluator, 0, len(t.keys))
		for _, key := range t.keys {
			resolvers = append(resolvers, expr.NewDotExpr(key))
		}
```